tests/e2e/multi_vc_utils.go

```diff
@@ -0,0 +1,433 @@
+
+	return allowedTopologiesMap
+}
+
+/*
+If we have multiple statefulsets, deployment Pods, PVCs/PVs created on a given namespace and for performing
+cleanup of these multiple sts creation, deleteAllStatefulSetAndPVs is used
+*/
+func deleteAllStatefulSetAndPVs(ctx context.Context, c clientset.Interface, ns string) {
+	StatefulSetPoll := 10 * time.Second
+	StatefulSetTimeout := 10 * time.Minute
+	ssList, err := c.AppsV1().StatefulSets(ns).List(context.TODO(),
+		metav1.ListOptions{LabelSelector: labels.Everything().String()})
+	framework.ExpectNoError(err)
+	errList := []string{}
+	for i := range ssList.Items {
+		ss := &ssList.Items[i]
+		var err error
+		if ss, err = scaleStatefulSetPods(c, ss, 0); err != nil {
+			errList = append(errList, fmt.Sprintf("%v", err))
+		}
+		fss.WaitForStatusReplicas(c, ss, 0)
+		framework.Logf("Deleting statefulset %v", ss.Name)
+		if err := c.AppsV1().StatefulSets(ss.Namespace).Delete(context.TODO(), ss.Name,
+			metav1.DeleteOptions{OrphanDependents: new(bool)}); err != nil {
+			errList = append(errList, fmt.Sprintf("%v", err))
+		}
+	}
+	pvNames := sets.NewString()
+	pvcPollErr := wait.PollImmediate(StatefulSetPoll, StatefulSetTimeout, func() (bool, error) {
+		pvcList, err := c.CoreV1().PersistentVolumeClaims(ns).List(context.TODO(),
+			metav1.ListOptions{LabelSelector: labels.Everything().String()})
+		if err != nil {
+			framework.Logf("WARNING: Failed to list pvcs, retrying %v", err)
+			return false, nil
+		}
+		for _, pvc := range pvcList.Items {
+			pvNames.Insert(pvc.Spec.VolumeName)
+			framework.Logf("Deleting pvc: %v with volume %v", pvc.Name, pvc.Spec.VolumeName)
+			if err := c.CoreV1().PersistentVolumeClaims(ns).Delete(context.TODO(), pvc.Name,
+				metav1.DeleteOptions{}); err != nil {
+				return false, nil
+			}
+		}
+		return true, nil
+	})
+	if pvcPollErr != nil {
+		errList = append(errList, "Timeout waiting for pvc deletion.")
+	}
+
+	pollErr := wait.PollImmediate(StatefulSetPoll, StatefulSetTimeout, func() (bool, error) {
+		pvList, err := c.CoreV1().PersistentVolumes().List(context.TODO(),
+			metav1.ListOptions{LabelSelector: labels.Everything().String()})
+		if err != nil {
+			framework.Logf("WARNING: Failed to list pvs, retrying %v", err)
+			return false, nil
+		}
+		waitingFor := []string{}
+		for _, pv := range pvList.Items {
+			if pvNames.Has(pv.Name) {
+				waitingFor = append(waitingFor, fmt.Sprintf("%v: %+v", pv.Name, pv.Status))
+			}
+		}
+		if len(waitingFor) == 0 {
+			return true, nil
+		}
+		framework.Logf("Still waiting for pvs of statefulset to disappear:\n%v", strings.Join(waitingFor, "\n"))
+		return false, nil
+	})
+	if pollErr != nil {
+		errList = append(errList, "Timeout waiting for pv provisioner to delete pvs, this might mean the test leaked pvs.")
+
+	}
+	if len(errList) != 0 {
+		framework.ExpectNoError(fmt.Errorf("%v", strings.Join(errList, "\n")))
+	}
+
+	depList, err := c.AppsV1().Deployments(ns).List(
```

Comment:
Give a log stating we are deleting Deployments

Reply:
Ack